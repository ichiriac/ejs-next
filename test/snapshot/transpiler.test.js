/**
 * Copyright (C) 2019 Ioan CHIRIAC (MIT)
 * @authors https://github.com/ichiriac/ejs2/graphs/contributors
 * @url https://ejs.js.org
 */
"use strict";

const lexer = require('../../lib/lexer');
const transpiler = require('../../lib/transpile');

describe('Transpiler', () => {
  it('Simple case', () => {
    var program = transpiler(new lexer(), `
    <foo> <%%
    <%# comment %>
    <%_ if (foo == "%>\\"%>") { _%>
      <%= bar %>
    <% } %>
    </foo>
    `, { strict: false, localsName: 'foo' });
    expect(program).toMatchSnapshot();
  });
  it('Simple case / strict mode', () => {
    var program = transpiler(new lexer(), `
    <foo> <%%
    <%# comment %>
    <%_ if (foo == "%>\\"%>") { _%>
      <%= bar %>
    <% } %>
    </foo>
    `, { strict: true, localsName: 'fooBar' });
    expect(program).toMatchSnapshot();
  });
  it('Template with redirected output', () => {
    var program = transpiler(new lexer(), `
      <%- include('foo', {
          contents: function() {@ %>
            Tara GROS tonkin !!!
          <% @}
        })
      %>
    `, { strict: true, localsName: '_' });
    expect(program).toMatchSnapshot();
  });    
  it('More complex', () => {
    var program = transpiler(new lexer(), `
OK, so have fun! :D
-------------------
<%
    var fruits = ["Apple", "Pear", "Orange", "Lemon"]
      , random = " ".repeat(198).split("").map(x => Math.random())
      ;
%>

These fruits are amazing:
<%_ for(var i = 0; i < fruits.length; ++i) { _%>

  - <%= fruits[i] %>s
<%_ } _%>

Let's see some random numbers:

<% 
  random.forEach(function(c, i) {
%> <%= c.toFixed(10) + ((i + 1) % 6 === 0 ? "\\n": "") %><%
  });
%>

Hello <%= locals.name %> !
    `, { strict: false, localsName: 'foo' });
    expect(program).toMatchSnapshot();
  });
});