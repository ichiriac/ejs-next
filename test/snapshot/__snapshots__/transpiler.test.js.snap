// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Transpiler More complex 1`] = `
"if (!ejs || typeof ejs.context !== \\"function\\") {
	foo = ejs;
	if (typeof window !== \\"undefined\\") {
		ejs = window.ejs;
	} else if (typeof global !== \\"undefined\\") {
		ejs = global.ejs;
	} else {
		return Promise.reject(new Error(\\"EJS module is not loaded\\"));
	}
}
foo = ejs.context(foo, {\\"strict\\":false,\\"localsName\\":\\"foo\\"},\\"undefined\\");
with(foo) {
	echo(\`
OK, so have fun! :D
-------------------
\`);
	
    var fruits = [\\"Apple\\", \\"Pear\\", \\"Orange\\", \\"Lemon\\"]
      , random = \\" \\".repeat(198).split(\\"\\").map(x => Math.random())
      ;
;
	echo(\`

These fruits are amazing:
\`);
	 for(var i = 0; i < fruits.length; ++i) { ;
	echo(\`
  - \`);
	safeEcho( fruits[i] );
	echo(\`s
\`);
	 } ;
	echo(\`
Let's see some random numbers:

\`);
	 
  random.forEach(function(c, i) {
;
	echo(\` \`);
	safeEcho( c.toFixed(10) + ((i + 1) % 6 === 0 ? \\"\\\\n\\": \\"\\") );
	
  });
;
	echo(\`

Hello \`);
	safeEcho( locals.name );
	echo(\` !
    \`);
}
return foo.resolveOutput();"
`;

exports[`Transpiler Simple case 1`] = `
"if (!ejs || typeof ejs.context !== \\"function\\") {
	foo = ejs;
	if (typeof window !== \\"undefined\\") {
		ejs = window.ejs;
	} else if (typeof global !== \\"undefined\\") {
		ejs = global.ejs;
	} else {
		return Promise.reject(new Error(\\"EJS module is not loaded\\"));
	}
}
foo = ejs.context(foo, {\\"strict\\":false,\\"localsName\\":\\"foo\\"},\\"undefined\\");
with(foo) {
	echo(\`
    <foo> \`);
	%
    <%# comment ;
	echo(\`
\`);
	 if (foo == \\";
	echo(\`\\\\\\"%>\\") { _%>
      \`);
	safeEcho( bar );
	echo(\`
    \`);
	 } ;
	echo(\`
    </foo>
    \`);
}
return foo.resolveOutput();"
`;
